eth/protocols/snap/gentrie.go

```diff
@@ -280,8 +280,9 @@ func (t *hashTrie) update(key, value []byte) error {
 
 // commit implements genTrie interface, committing the nodes on right boundary.
 func (t *hashTrie) commit(complete bool) common.Hash {
+	hash := t.tr.Hash() // flush the nodes on right boundary
```

Comment:
For the stacktrie, this will forcibly "The main purpose here is to commit the nodes on right boundary", and this make it so that it's not possible to add more data to this stacktrie. (because there will be no nodes to insert into, only one hashed rootnodes). 
However, this method is used in places where there's more data to come, e..g 
```golang
	} else if task.genBatch.ValueSize() > batchSizeThreshold {
		task.genTrie.commit(false)
```
A "time to push to disk for now" invocation^ 

There are two concerns, but only one flag
- Caller wants to commit to release a bit of memory, 
- Caller wants to force-commit to flush right-hand boundary
-

The stacktrie doesn't really _need_ "commit to release some memory", since it does so continuously. Perhaps we need two flags